### PR TITLE
Setup eslint and fix problems

### LINKED ISSUE #11: Set up linter for this repository
the linter should be compliant with that of brave-core

### PR DESCRIPTION
**Pull request by sangwoo108**
fix #11 

Most of the rules are brought from brave-core.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        ],
+        'altHeaders': [
+          [
+            'Copyright (c) {YEAR} The Brave Authors. All rights reserved.',
+            '* This Source Code Form is subject to the terms of the Mozilla Public',
+            '* License, v. 2.0. If a copy of the MPL was not distributed with this file,',
+            '* You can obtain one at https://mozilla.org/MPL/2.0/.'
+          ],
+          [
+            'This Source Code Form is subject to the terms of the Mozilla Public',
```
Comment (fallaciousreasoning) [inline .eslintrc.js]: You can remove this one, if all your headers use the proper format. It's only for legacy purposes in Brave-Core

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            '* License, v. 2.0. If a copy of the MPL was not distributed with this file,',
+            '* You can obtain one at https://mozilla.org/MPL/2.0/.'
+          ],
+          [
+            'Copyright (c) {YEAR} The Brave Authors. All rights reserved.',
```
Comment (fallaciousreasoning) [inline .eslintrc.js]: This too

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        },
+        'header': [
+          'Copyright (c) {YEAR} The Brave Authors. All rights reserved.',
+          'This Source Code Form is subject to the terms of the Mozilla Public',
+          'License, v. 2.0. If a copy of the MPL was not distributed with this file,',
+          'You can obtain one at https://mozilla.org/MPL/2.0/.'
+        ],
+        'altHeaders': [
+          [
+            'Copyright (c) {YEAR} The Brave Authors. All rights reserved.',
```
Comment (fallaciousreasoning) [inline .eslintrc.js]: This too.

File: youtube.com.js [hunk 4]
```diff
@@ -11,23 +11,23 @@ const siteSpecificDetector = { // eslint-disable-line no-unused-vars
       // The last one is known to be one with the highest resolution
       thumbnail = thumbnail[thumbnail.length - 1]?.url
     }
-    return (typeof(thumbnail) === 'string' && (thumbnail.startsWith('/') || thumbnail.startsWith('http://') || thumbnail.startsWith('https://'))) 
-        ? thumbnail : null
+    return (typeof (thumbnail) === 'string' && (thumbnail.startsWith('/') || thumbnail.startsWith('http://') || thumbnail.startsWith('https://')))
```
Comment (fallaciousreasoning) [inline youtube.com.js]: This is just an observation but I don't think we normally add the brackets around

```ts
// prefer
typeof thing === 'type'
// over
typeof (thing) === 'type'
```

But it's not in the style guide, and it's not your code so this is fine :laughing:

**Review by fallaciousreasoning — APPROVED** [review]
This LGTM

File: youtube.com.js (hunk 4 above)
Comment (sangwoo108) [reply youtube.com.js]: Oh, yeah. I always forget it 🤣

File: .eslintrc.js (hunk 1 above)
Comment (sangwoo108) [reply .eslintrc.js]: Cool! Done.

**Comment by sangwoo108** [comment]
@SocketSecurity ignore eslint-plugin-licenses@1.0.2

**Pull request merged**